.github/workflows/builder_nodejs_slsa3.yml

```diff
@@ -108,35 +95,26 @@ jobs:
         id: generate
         uses: slsa-framework/slsa-github-generator/actions/delegator/setup-token@main
         with:
-          slsa-workflow-recipient: "delegator_generic_slsa3.yml"
+          slsa-workflow-recipient: "delegator_lowperms-generic_slsa3.yml"
           slsa-rekor-log-public: ${{ inputs.rekor-log-public }}
           slsa-runner-label: "ubuntu-latest"
           slsa-build-action-path: "./internal/nodejs-action"
           slsa-version: "v0.2"
-          # TODO(#1575): mask sensitive fields.
           slsa-workflow-inputs: ${{ toJson(inputs) }}
 
   slsa-run:
     needs: [slsa-setup]
     permissions:
       id-token: write # For signing.
-      contents: write # For asset uploads.
-      packages: write # For publishing to GitHub packages.
-    uses: slsa-framework/slsa-github-generator/.github/workflows/delegator_generic_slsa3.yml@main
+      contents: read # For repo checkout.
+      actions: read # For getting workflow run on private repos.
```

Comment:
i don't know for sure which perm is needed to get the run https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/actions/verify-token/src/predicate02.ts#L38

What's not clear to me is why we need to get use this API to get the run https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/actions/verify-token/src/predicate02.ts#L77C12-L79

Actor name and id are present in the env variables `GITHUB_ACTOR` and `GITHUB_ACTOR_ID`, so we should be able to get them for free without permissions?

Reply:
> i don't know for sure which perm is needed to get the run https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/actions/verify-token/src/predicate02.ts#L38
The permission required is documented here:
https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#actions

I was able to test that adding this permission works for a private repo and removing it causes the request to fail.

> What's not clear to me is why we need to use this API to get the run https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/actions/verify-token/src/predicate02.ts#L77C12-L79
> 
> Actor name and id are present in the env variables `GITHUB_ACTOR` and `GITHUB_ACTOR_ID`, so we should be able to get them for free without permissions and without calling the API?

triggering actor id is apparently the actor who triggered the re-run. So user A could trigger a workflow run, and then user B could trigger the re-run. The actor_id would be user A and triggering_user_id would be user B in that case. They currently don't have an env var or github context field for the triggering_actor_id though there is a triggering_actor (username) on the github context.

Mark seemed to think it was important enough to include. 
https://github.com/slsa-framework/github-actions-buildtypes/blob/bc8206c3c0f38e4925285ac9091a18431b010783/workflow/v1/README.md#internal-parameters

I suppose that we could just wait to include it until GHA provides an env var or a github context field but `actions: read` does give us the flexibility to get it if we need to (we also just seem to have it by default for public repos so it's not a lot of extra permissions).

Reply:
gotcha. Did not realize triggering actor was not set on re-run.